### PR TITLE
MAPPA - Animation tools

### PR DESCRIPTION
**Pull request by patreeceeo**
## Summary

_A summary of changes being made in this PR_

## References

closes #_insert number here_

## Checklist
- [ ] If this PR is still a WIP, convert to a draft
- [ ] [ensure all checks pass](https://github.com/XRFoundation/XREngine/wiki/Testing-&-Contributing)
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewer

## QA Steps

_List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos._

### CODE REVIEW

**Comment by HexaField** [comment]
The first problem I see here is replacing the default avatars. This should be done via removing the default project, and creating your own, rather than overriding the base engine.

More generally, it would be better to see application specific features (UI, game logic, custom assets) implemented via a project rather than the base engine.

Project API can be found here https://github.com/XRFoundation/ethereal-engine-docs/blob/master/docs/3_concepts/1_projects_api.md

File: packages/engine/src/scene/systems/TriggerSystem.ts [hunk 1]
```diff
@@ -1,41 +1,81 @@
 import { Entity } from '../../ecs/classes/Entity'
```
Comment (HexaField) [inline packages/engine/src/scene/systems/TriggerSystem.ts:1]: the changes to this system must be reverted if this branch is to be merged, the existing functionality of triggers should cover all use cases. Please re-think the implementation here.

File: packages/engine/src/scene/functions/loaders/ModelFunctions.ts [hunk 2]
```diff
@@ -42,6 +44,27 @@ export const deserializeModel: ComponentDeserializeFunction = (entity: Entity, d
   setComponent(entity, SceneAssetPendingTagComponent, true)
 }
 
+function setupAnimations(model, scene, entity) {
```
Comment (HexaField) [inline packages/engine/src/scene/functions/loaders/ModelFunctions.ts:47]: what is the reason for doing this? this should be done in the existing LoopAnimationFunctions implementation

**Review by HexaField — CHANGES_REQUESTED** [review]

**Comment by patreeceeo** [comment]
Sorry, didn't mean to close this. Will reopen.

**Comment by patreeceeo** [comment]
@AidanCaruso take a look at Josh's comments when you have a second.

**Pull request closed without merging**